crucible-saw/src/Lang/Crucible/Backend/SAWCore.hs

```diff
@@ -962,6 +962,18 @@ evaluateExpr sym sc cache = f
            --     1 here to make things match up.
            n <- SC.scNat sc (w - 1)
            join (SC.scBvSShr sc n <$> f x <*> (SC.scBvToNat sc w =<< f y))
+        B.BVRol w x y -> fmap SAWExpr $ do
```

Comment:
Another option would be to use the saw-core prelude operations `bvRotateL` and `bvRotateR` here instead of `rotateL` and `rotateR`. They are defined in terms of `rotate{L,R}` together with `bvToNat`, so it probably doesn't actually matter much.
```
bvRotateL : (n : Nat) -> (a : sort 0) -> (w : Nat) -> Vec n a -> bitvector w -> Vec n a;
bvRotateL n a w xs i = rotateL n a xs (bvToNat w i);

bvRotateR : (n : Nat) -> (a : sort 0) -> (w : Nat) -> Vec n a -> bitvector w -> Vec n a;
bvRotateR n a w xs i = rotateR n a xs (bvToNat w i);
```

Reply:
I guess it really only makes a difference if there are rewrites that want to specifically target `bvRotate*`...

What's your preference here?

Reply:
I don't think we have any rewrites on `bvRotate*` (in fact, `rotate*` probably are a better form for rewriting). So we might as well leave this bit as it is.

Reply:
OK, we'll leave it like this.